### ISSUE TITLE
Game crashes when I try to open the door with a key

### ISSUE THREAD

**Issue by antonte** [frogatto/frogatto#314]
ERROR: asserts.cpp:149 : ASSERT FAIL: variant_type.cpp:2145 ASSERTION FAILED: INVALID TOKEN WHEN PARSING TYPE: particle_color_names AT:
At modules/frogatto/data/objects/effects/general/bouncing_debris_chunk.cfg 154:
particle_color_names
^

CRITICAL: At modules/frogatto/data/objects/props-interactive/locks/padlock.cfg 27:
[ level.player.store_event(event_name, 1),
^

---
CRITICAL: stack trace:
CRITICAL:   /home/anton/prj/third_party/anura/anura : fatal_assert_failure_exception::fatal_assert_failure_exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x2fd
CRITICAL:   /home/anton/prj/third_party/anura/anura : parse_variant_type(variant const&, formula_tokenizer::Token const*&, formula_tokenizer::Token const*, bool)+0x7e23
CRITICAL:   /home/anton/prj/third_party/anura/anura : parse_variant_type(variant const&)+0xf3
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x11019f2]
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::CustomObjectType(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, variant, CustomObjectType const*, CustomObjectType const*)+0x3fe3
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::initSubObjects(variant, CustomObjectType const*)+0x745
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::CustomObjectType(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, variant, CustomObjectType const*, CustomObjectType const*)+0x411c
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::recreate(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CustomObjectType const*)+0x2a2
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x291
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::getOrDie(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x29
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::CustomObject(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, bool)+0x6b
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x1240d77]
CRITICAL:   /home/anton/prj/third_party/anura/anura : formula_vm::VirtualMachine::executeInternal(game_logic::FormulaCallable const&, std::vector<ffl::IntrusivePtr<game_logic::FormulaCallable>, std::allocator<ffl::IntrusivePtr<game_logic::FormulaCallable> > >&, std::vector<variant, std::allocator<variant> >&, std::vector<variant, std::allocator<variant> >&, short const*, short const*) const+0x4ce5
CRITICAL:   /home/anton/prj/third_party/anura/anura : formula_vm::VirtualMachine::executeInternal(game_logic::FormulaCallable const&, std::vector<ffl::IntrusivePtr<game_logic::FormulaCallable>, std::allocator<ffl::IntrusivePtr<game_logic::FormulaCallable> > >&, std::vector<variant, std::allocator<variant> >&, std::vector<variant, std::allocator<variant> >&, short const*, short const*) const+0x43fc
CRITICAL:   /home/anton/prj/third_party/anura/anura : formula_vm::VirtualMachine::execute(game_logic::FormulaCallable const&) const+0xaf
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0xb9c471]
CRITICAL:   /home/anton/prj/third_party/anura/anura : game_logic::Formula::execute(game_logic::FormulaCallable const&) const+0x9e
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEventInternal(int, game_logic::FormulaCallable const*, bool)+0x3ea
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEvent(int, game_logic::FormulaCallable const*)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x1200af1]
CRITICAL:   /home/anton/prj/third_party/anura/anura : EntityCommandCallable::runCommand(Level&, Entity&) const+0x4a
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0x164
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::resolveDelayedEvents()+0x2d
CRITICAL:   /home/anton/prj/third_party/anura/anura : detect_user_collisions(Level&)+0xae3
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::do_processing()+0x5a
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::process()+0x1a2
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_cycle()+0x2382
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_level()+0xa4
CRITICAL:   /home/anton/prj/third_party/anura/anura : main()+0x7982
CRITICAL:   /lib/x86_64-linux-gnu/libc.so.6 : __libc_start_main()+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : _start()+0x29
CRITICAL: ---
CRITICAL: custom_object_functions.cpp:114 ASSERTION FAILED: variant_type.cpp:2145 ASSERTION FAILED: INVALID TOKEN WHEN PARSING TYPE: particle_color_names AT:
At modules/frogatto/data/objects/effects/general/bouncing_debris_chunk.cfg 154:
particle_color_names
^

ERROR ENCOUNTERED WHILE RUNNING COMMAND GENERATED BY THIS EXPRESSION:
At modules/frogatto/data/object_prototypes/door.cfg 12:
[fire_event(lock, 'unlock'), 
 ^-------------------------^

CRITICAL: 
---
CRITICAL: stack trace:
CRITICAL:   /home/anton/prj/third_party/anura/anura : EntityCommandCallable::runCommand(Level&, Entity&) const+0x24f
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0x164
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::resolveDelayedEvents()+0x2d
CRITICAL:   /home/anton/prj/third_party/anura/anura : detect_user_collisions(Level&)+0xae3
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::do_processing()+0x5a
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::process()+0x1a2
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_cycle()+0x2382
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_level()+0xa4
CRITICAL:   /home/anton/prj/third_party/anura/anura : main()+0x7982
CRITICAL:   /lib/x86_64-linux-gnu/libc.so.6 : __libc_start_main()+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : _start()+0x29
CRITICAL: ---
INFO: asserts.cpp:74 : ATTEMPTING TO SEND CRASH REPORT...
[Thread 0x7fffed9ba700 (LWP 18612) exited]

Thread 1 "anura" received signal SIGABRT, Aborted.
0x00007ffff59da269 in raise (sig=6) at ../sysdeps/unix/sysv/linux/pt-raise.c:35
35	../sysdeps/unix/sysv/linux/pt-raise.c: No such file or directory.
(gdb)

**Comment by antonte** [comment]
Still crashing:

ERROR: asserts.cpp:149 : ASSERT FAIL: variant_type.cpp:2145 ASSERTION FAILED: INVALID TOKEN WHEN PARSING TYPE: particleColorNames AT:
At modules/frogatto/data/objects/effects/general/bouncing_debris_chunk.cfg 154:
particleColorNames
^


CRITICAL: At modules/frogatto/data/objects/props-interactive/locks/padlock.cfg 27:
[ level.player.store_event(event_name, 1),
^


---
CRITICAL: stack trace:
CRITICAL:   /home/anton/prj/third_party/anura/anura : fatal_assert_failure_exception::fatal_assert_failure_exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x2fd
CRITICAL:   /home/anton/prj/third_party/anura/anura : parse_variant_type(variant const&, formula_tokenizer::Token const*&, formula_tokenizer::Token const*, bool)+0x7e23
CRITICAL:   /home/anton/prj/third_party/anura/anura : parse_variant_type(variant const&)+0xf3
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x11019f2]
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::CustomObjectType(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, variant, CustomObjectType const*, CustomObjectType const*)+0x3fe3
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::initSubObjects(variant, CustomObjectType const*)+0x745
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::CustomObjectType(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, variant, CustomObjectType const*, CustomObjectType const*)+0x411c
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::recreate(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CustomObjectType const*)+0x2a2
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x291
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObjectType::getOrDie(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x29
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::CustomObject(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, bool)+0x6b
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x1240d77]
CRITICAL:   /home/anton/prj/third_party/anura/anura : formula_vm::VirtualMachine::executeInternal(game_logic::FormulaCallable const&, std::vector<ffl::IntrusivePtr<game_logic::FormulaCallable>, std::allocator<ffl::IntrusivePtr<game_logic::FormulaCallable> > >&, std::vector<variant, std::allocator<variant> >&, std::vector<variant, std::allocator<variant> >&, short const*, short const*) const+0x4ce5
CRITICAL:   /home/anton/prj/third_party/anura/anura : formula_vm::VirtualMachine::executeInternal(game_logic::FormulaCallable const&, std::vector<ffl::IntrusivePtr<game_logic::FormulaCallable>, std::allocator<ffl::IntrusivePtr<game_logic::FormulaCallable> > >&, std::vector<variant, std::allocator<variant> >&, std::vector<variant, std::allocator<variant> >&, short const*, short const*) const+0x43fc
CRITICAL:   /home/anton/prj/third_party/anura/anura : formula_vm::VirtualMachine::execute(game_logic::FormulaCallable const&) const+0xaf
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0xb9c471]
CRITICAL:   /home/anton/prj/third_party/anura/anura : game_logic::Formula::execute(game_logic::FormulaCallable const&) const+0x9e
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEventInternal(int, game_logic::FormulaCallable const*, bool)+0x3ea
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEvent(int, game_logic::FormulaCallable const*)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x1200af1]
CRITICAL:   /home/anton/prj/third_party/anura/anura : EntityCommandCallable::runCommand(Level&, Entity&) const+0x4a
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0x164
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEventInternal(int, game_logic::FormulaCallable const*, bool)+0x336
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEvent(int, game_logic::FormulaCallable const*)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x1200af1]
CRITICAL:   /home/anton/prj/third_party/anura/anura : EntityCommandCallable::runCommand(Level&, Entity&) const+0x4a
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0x164
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEventInternal(int, game_logic::FormulaCallable const*, bool)+0x336
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEvent(int, game_logic::FormulaCallable const*)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura : PlayableCustomObject::process(Level&)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::do_processing()+0x2ee
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::process()+0x1a2
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_cycle()+0x2382
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_level()+0xa4
CRITICAL:   /home/anton/prj/third_party/anura/anura : main()+0x7982
CRITICAL:   /lib/x86_64-linux-gnu/libc.so.6 : __libc_start_main()+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : _start()+0x29
CRITICAL: ---
CRITICAL: custom_object_functions.cpp:114 ASSERTION FAILED: variant_type.cpp:2145 ASSERTION FAILED: INVALID TOKEN WHEN PARSING TYPE: particleColorNames AT:
At modules/frogatto/data/objects/effects/general/bouncing_debris_chunk.cfg 154:
particleColorNames
^


ERROR ENCOUNTERED WHILE RUNNING COMMAND GENERATED BY THIS EXPRESSION:
At modules/frogatto/data/object_prototypes/door.cfg 12:
[fire_event(lock, 'unlock'), 
 ^-------------------------^


CRITICAL: 
---
CRITICAL: stack trace:
CRITICAL:   /home/anton/prj/third_party/anura/anura : EntityCommandCallable::runCommand(Level&, Entity&) const+0x24f
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0x164
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEventInternal(int, game_logic::FormulaCallable const*, bool)+0x336
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEvent(int, game_logic::FormulaCallable const*)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura() [0x1200af1]
CRITICAL:   /home/anton/prj/third_party/anura/anura : EntityCommandCallable::runCommand(Level&, Entity&) const+0x4a
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0x164
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::executeCommand(variant const&)+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEventInternal(int, game_logic::FormulaCallable const*, bool)+0x336
CRITICAL:   /home/anton/prj/third_party/anura/anura : CustomObject::handleEvent(int, game_logic::FormulaCallable const*)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura : PlayableCustomObject::process(Level&)+0x185
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::do_processing()+0x2ee
CRITICAL:   /home/anton/prj/third_party/anura/anura : Level::process()+0x1a2
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_cycle()+0x2382
CRITICAL:   /home/anton/prj/third_party/anura/anura : LevelRunner::play_level()+0xa4
CRITICAL:   /home/anton/prj/third_party/anura/anura : main()+0x7982
CRITICAL:   /lib/x86_64-linux-gnu/libc.so.6 : __libc_start_main()+0xf0
CRITICAL:   /home/anton/prj/third_party/anura/anura : _start()+0x29
CRITICAL: ---
INFO: asserts.cpp:74 : ATTEMPTING TO SEND CRASH REPORT...

**Comment by rkettering** [comment]
Yeah, that was a very-late-in-the-evening "fix" I did in a hurry before I went to bed, sorry about that - I just typed in some code and didn't have time to test it, because my copy of Anura was out of date, and I didn't have enough time to recompile the engine.

The issue was that local type signatures like that have a requirement that they be in a specific kind of "CamelCase" - the kind where the the first letter is also capitalized.  This requirement is something we're imitating from Haskell, afaik.

Thanks again for catching this. :)
